examples/rl/pendulum/ex_save_and_load.py

```diff
@@ -0,0 +1,157 @@
+
+# isort: on
+
+import jax
+from brax import envs
+from brax.io import model
+from brax.training import distribution
+from brax.training.acme import running_statistics
+from brax.training.agents.ppo import train as ppo
+from brax.training.agents.ppo.networks import make_inference_fn
+from mujoco import mjx
+
+from ambersim.rl.networks import MLP, BraxPPONetworksWrapper
+from ambersim.rl.pendulum.swingup import PendulumSwingupEnv
+
+"""
+Perform pendulum swingup training with custom policy architectures, and
+demonstrate saving and loading the trained networks from disk.
+"""
+
+
+def train_swingup():
+    """Train a pendulum swingup agent with custom network architectures."""
+    # Initialize the environment
+    envs.register_environment("pendulum_swingup", PendulumSwingupEnv)
+    env = envs.get_environment("pendulum_swingup")
+
+    # Create some funky custom network architectures
+    policy_network = MLP(layer_sizes=(512, 4, 128, 2))  # outputs mean and variance
+    value_network = MLP(layer_sizes=(26, 128, 53, 1))  # outputs a single value
+    network_wrapper = BraxPPONetworksWrapper(
+        policy_network=policy_network,
+        value_network=value_network,
+        action_distribution=distribution.NormalTanhDistribution,
+    )
+
+    train_fn = functools.partial(
+        ppo.train,
+        num_timesteps=100_000,
+        num_evals=50,
+        reward_scaling=0.1,
+        episode_length=200,
+        normalize_observations=True,
+        action_repeat=1,
+        unroll_length=10,
+        num_minibatches=32,
+        num_updates_per_batch=8,
+        discounting=0.97,
+        learning_rate=3e-4,
+        entropy_cost=0,
+        num_envs=1024,
+        batch_size=512,
+        network_factory=network_wrapper.make_ppo_networks,
+        seed=0,
+    )
+
+    # Define a callback to log progress
+    times = [datetime.now()]
+
+    def progress(num_steps, metrics):
+        """Logs progress during RL."""
+        print(f"  Steps: {num_steps}, Reward: {metrics['eval/episode_reward']}")
+        times.append(datetime.now())
+
+    # Do the training
+    print("Training...")
+    make_inference_fn, params, _ = train_fn(
+        environment=env,
+        progress_fn=progress,
+    )
+
+    print(f"Time to jit: {times[1] - times[0]}")
+    print(f"Time to train: {times[-1] - times[1]}")
+
+    # Save both the parameters and the networks to disk
+    print("Saving...")
+    params_path = "/tmp/pendulum_params.pkl"
+    networks_path = "/tmp/pendulum_networks.pkl"
+    model.save_params(params_path, params)
+    with open(networks_path, "wb") as f:
+        pickle.dump(network_wrapper, f)
+
+
+def test_trained_swingup_policy():
+    """Load a trained policy and run an interactive simulation."""
+    envs.register_environment("pendulum_swingup", PendulumSwingupEnv)
+    env = envs.get_environment("pendulum_swingup")
+    mj_model = env.model
+    mj_data = mujoco.MjData(mj_model)
+    obs = env.compute_obs(mjx.device_put(mj_data), {})
```

Comment:
Should we also consider forcing our custom envs to have a util that returns this boilerplate stuff?

Reply:
Ditto with the comment above and below - could definitely be worthwhile but in a separate PR